### ISSUE TITLE
Lack of newline breaks Module::Metadata

### ISSUE THREAD

**Issue by timekillerj** [damil/Text-Transliterator#1]
In Unaccent.pm the VERSION string causes a failure if Text::Transliterator is installed to a local lib.

This can be seen if you enter the following:

cpanm --local-lib=./local Search::Tokenizer

The above line tries to install Search::Tokenizer into ./local. Search::Tokenizer has Text::Transliterator as a dependancy. Text::Transliterator gets installed fine, but when cpanm uses Module::Metadata to determine the installed version of Text::Transliterator, the following error is produced:

Error evaling version line 'BEGIN { my $dummy = q#  Hide from _packages_inside()
    #; package Module::Metadata::_version::p39;
    use version;
    no strict;
    no warnings;

```
  $vsub = sub {
    local $VERSION;
    $VERSION=undef;
    use Text::Transliterator; our $VERSION = $Text::Transliterator::VERSION;;
    $VERSION
  };
```

  }' in /home/jdonahue/local/lib/perl5/Text/Transliterator/Unaccent.pm: Can't locate Text/Transliterator.pm in @INC (you may need to install the Text::Transliterator module) (@INC contains: lib FatPacked::35044624=HASH(0x216bd10) /etc/perl /usr/local/lib/x86_64-linux-gnu/perl/5.20.1 /usr/local/share/perl/5.20.1 /usr/lib/x86_64-linux-gnu/perl5/5.20 /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.20 /usr/share/perl/5.20 /usr/local/lib/site_perl .) at (eval 104) line 10, <$handle> line 6.
BEGIN failed--compilation aborted at (eval 104) line 10, <$handle> line 6.

failed to build version sub for /home/jdonahue/local/lib/perl5/Text/Transliterator/Unaccent.pm at /usr/share/perl5/CPAN/Meta/Check.pm line 19.

Basically, since Text::Transliterator is installed to a local lib path, it can't be found. this causes the script to abort. If instead of:

 `use Text::Transliterator; our $VERSION = $Text::Transliterator::VERSION;`

you added a newline:

`use Text::Transliterator; 
our $VERSION = $Text::Transliterator::VERSION;'

Then the line passed to Module Metadata would just be `our $VERSION = $Text::Transliterator::VERSION;` - which will eval to undef since Text::Transliterator still can't be found, but the script will not error out and the modules will be successfully installed.

I know the ideal solution is for Module::Metadata to handle this better, and we will be filing a bug report with them as well, but I was hoping the newline would be a simple enough change that you could add it.

**Comment by damil** [comment]
Le 06.11.2014 19:44, Jason Donahue a écrit :

> In Unaccent.pm the VERSION string causes a failure if 
> Text::Transliterator is installed to a local lib.
> 
> This can be seen if you enter the following:
> 
> cpanm --local-lib=./local Search::Tokenizer
> 
> The above line tries to install Search::Tokenizer into ./local. 
> Search::Tokenizer has Text::Transliterator as a dependancy. 
> Text::Transliterator gets installed fine, but when cpanm uses 
> Module::Metadata to determine the installed version of 
> Text::Transliterator, the following error is produced:
> 
> Error evaling version line 'BEGIN { my $dummy = q# Hide from 
> _packages_inside()
> #; package Module::Metadata::_version::p39;
> use version;
> no strict;
> no warnings;
> 
> |   $vsub = sub {
>      local $VERSION;
>      $VERSION=undef;
>      use Text::Transliterator; our $VERSION = $Text::Transliterator::VERSION;;
>      $VERSION
>    };
> |
> 
> }' in /home/jdonahue/local/lib/perl5/Text/Transliterator/Unaccent.pm: 
> Can't locate Text/Transliterator.pm in @INC https://github.com/INC 
> (you may need to install the Text::Transliterator module) (@INC 
> https://github.com/INC contains: lib 
> FatPacked::35044624=HASH(0x216bd10) /etc/perl 
> /usr/local/lib/x86_64-linux-gnu/perl/5.20.1 
> /usr/local/share/perl/5.20.1 /usr/lib/x86_64-linux-gnu/perl5/5.20 
> /usr/share/perl5 /usr/lib/x86_64-linux-gnu/perl/5.20 
> /usr/share/perl/5.20 /usr/local/lib/site_perl .) at (eval 104) line 
> 10, <$handle> line 6.
> BEGIN failed--compilation aborted at (eval 104) line 10, <$handle> line 6.
> 
> failed to build version sub for 
> /home/jdonahue/local/lib/perl5/Text/Transliterator/Unaccent.pm at 
> /usr/share/perl5/CPAN/Meta/Check.pm line 19.
> 
> Basically, since Text::Transliterator is installed to a local lib 
> path, it can't be found. this causes the script to abort. If instead of:
> 
> |use Text::Transliterator; our $VERSION = $Text::Transliterator::VERSION;|
> 
> you added a newline:
> 
> `use Text::Transliterator;
> our $VERSION = $Text::Transliterator::VERSION;'
> 
> Then the line passed to Module Metadata would just be |our $VERSION = 
> $Text::Transliterator::VERSION;| - which will eval to undef since 
> Text::Transliterator still can't be found, but the script will not 
> error out and the modules will be successfully installed.
> 
> I know the ideal solution is for Module::Metadata to handle this 
> better, and we will be filing a bug report with them as well, but I 
> was hoping the newline would be a simple enough change that you could 
> add it.
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/damil/Text-Transliterator/issues/1.
> 
> Hi Jason,

Following your suggestion would perhaps make Module::Metadata happy, but 
it would break other parts of the CPAN toolchain; in particular, the 
convention is that the code setting $VERSION should stand on one single 
line (see http://www.dagolden.com/index.php/2191/real-versions-on-cpan/ ).

It seems to me that there is a problem between your @INC and the local 
directory you are installing into. Maybe editing PERL5LIB or adding a 
-Mlocal_dir option would solve that.

Laurent D.
